src/scan/postgres_scan.cpp

```diff
@@ -39,9 +39,25 @@ PostgresScanGlobalState::InitGlobalState(duckdb::TableFunctionInitInput &input)
 		return;
 	}
 
-	/* We need ordered columns ids for reading tuple. */
+	/* We need ordered columns id for reading tuple. */
+	duckdb::map<duckdb::idx_t, duckdb::column_t> ordered_input_columns;
 	for (duckdb::idx_t i = 0; i < input.column_ids.size(); i++) {
-		m_read_columns_ids[input.column_ids[i]] = i;
+		ordered_input_columns[input.column_ids[i]] = i;
+	}
+
+	auto table_filters = input.filters.get();
+	m_column_filters.resize(input.column_ids.size(), 0);
+
+	for (auto const &[column_id, output_idx] : ordered_input_columns) {
+		duckdb::TableFilter *column_filter = nullptr;
+		if (table_filters) {
+			auto column_filter_it = table_filters->filters.find(output_idx);
+			if (column_filter_it != table_filters->filters.end()) {
+				column_filter = column_filter_it->second.get();
+			}
+		}
+		m_input_columns.push_back(duckdb::make_pair(column_id, output_idx));
```

Comment:
nit:
```suggestion
		m_input_columns.emplace_back(column_id, output_idx);
```
(same below) for `m_output_columns`